firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/messages.txt

```diff
@@ -0,0 +1,207 @@
+      THEN("The write method reports invalid type status")
+      THEN("After the write method is called, the type member remains unchanged")
+      THEN("The payload.values field data remains unchanged")
+      THEN("The output buffer remains empty")
+
+  GIVEN("A Message object default-constructed with StateSegment Taggedunion, default message types and a max size of 126 bytes as its class template parameters")
+    // setup:
+    // sensor_measurements.flow = 30;
+    // sensor_measurements.fio2 = 85;
+    // sensor_measurements.spo2 = 72;  
+    WHEN("The payload is a sensor measurements message and write method is called")
+      THEN("The write method reports ok status")
+      THEN("After the write method is called, the type member is set to the value of the message payload tag")
+      THEN("The type field of the body's header correctly stores the value of message type obtained from the message payload tag")
+      THEN("The output buffer has an expected sequence of 16 bytes")
+
+  GIVEN("A Message object constructed with StateSegment Taggedunion, default message types and a max size of 508 bytes")
+    // setup:
+    // parameters.fio2 = 60;
+    // parameters.mode = VentilationMode_hfnc;
+    // parameters.ventilating = true;
+    WHEN("The parameters message data is written")
+      THEN("The write method reports ok status")
+      THEN("After the write method is called, the type member is set to the value of the message payload tag")
+      THEN("The type field of the body's header correctly stores the value of message type obtained from the message payload tag")
+      THEN("The output buffer has an expected sequence of 10 bytes")
+
+/* Write function for each of the message types (example: sensor_measurements)
+    // setup:
+    // sensor_measurements.flow = 30;
+    // sensor_measurements.fio2 = 85;
+    // sensor_measurements.spo2 = 72;
+    WHEN("The payload is a sensor measurements message and write method is called")
+      THEN("The write method reports ok status")
+      THEN("After the write method is called, the type member is equal to the value of sensor_measurements tag")
+      THEN("The type field of the body's header correctly stores the value of message type obtained from the message payload tag")
+      THEN("The output buffer has an expected sequence of 16 bytes")
+/*
+
+/* Parse function
+
+Scenario: Message: The parse method correctly updates internal message type and sets the message payload fields after decoding from the input buffer
+
+  GIVEN("A Message object constructed with StateSegment Taggedunion, default message types and a max size of 252 bytes")
+    // setup:
+    // message class type member is unchanged
+    // payload is uninitialised
+
+    WHEN("An empty input buffer body is parsed")
+      THEN("The parse method reports invalid length status")
+      THEN("The payload Taggedunion tag is set to Unknown")
+      THEN("the payload.values field remains unchanged")
```

Comment:
What happens to the `type` field after the method call? (This applies to all parse and MessageReceiver tests)

Reply:
fixed.